mne/decoding/csp.py

```diff
@@ -156,87 +173,24 @@ def fit(self, X, y):
         self : instance of CSP
             Returns the modified instance.
         """
-        if not isinstance(X, np.ndarray):
-            raise ValueError("X should be of type ndarray (got %s)."
-                             % type(X))
         self._check_Xy(X, y)
-        n_channels = X.shape[1]
 
         self._classes = np.unique(y)
         n_classes = len(self._classes)
         if n_classes < 2:
             raise ValueError("n_classes must be >= 2.")
+        if n_classes > 2 and self.component_order == 'alternate':
+            raise ValueError("component_order='alternate' requires two "
+                             "classes, but data contains {} classes; use "
+                             "component_order='mutual_info' "
+                             "instead.".format(n_classes))
```

Comment:
codecov complains, easy to add some `pytest.raises` to a test?

Reply:
Done

Reply:
So did I, but did not get a merge conflict... did you @cbrnr ?